src/modules/ims_isc/ims_isc_mod.c

```diff
@@ -264,16 +262,27 @@ int isc_match_filter(struct sip_msg *msg, char *str1, udomain_t *d)
 	/* starting or resuming? */
 	memset(&old_mark, 0, sizeof(isc_mark));
 	memset(&new_mark, 0, sizeof(isc_mark));
+	memset(&dummy_mark, 0, sizeof(isc_mark));
+
+	if(is_route_type(FAILURE_ROUTE)) {
+		status |= ISCMARK_FAILURE;
+	}
+
 	if(isc_mark_get_from_msg(msg, &old_mark)) {
 		LM_DBG("Message returned s=%d;h=%d;d=%d;a=%.*s\n", old_mark.skip,
 				old_mark.handling, old_mark.direction, old_mark.aor.len,
 				old_mark.aor.s);
+	} else if(is_route_type(FAILURE_ROUTE)
+			  && isc_mark_get_from_lumps(msg, &dummy_mark)) {
```

Comment:
@kilianwww Just a quick question here, I've seen that the dummy mark is never used anywhere else, how will the iFC execution continue with the correct index in the failure route?